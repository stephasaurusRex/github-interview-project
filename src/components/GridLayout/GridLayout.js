import React from "react";
import { Responsive, WidthProvider } from "react-grid-layout";
import styled from "styled-components";

import { index } from "../../styles";

const ResponsiveGridLayout = WidthProvider(Responsive);

const GridItem = styled.div`
  border: #000 1px solid;
  align-items: center;
  justify-content: center;
  display: flex;
`;

class GridLayout extends React.Component {
  render() {
    const layout = [
      { i: "a", x: 1, y: 0, w: 1, h: 1 },
      { i: "b", x: 2, y: 0, w: 1, h: 1 },
      { i: "c", x: 1, y: 3, w: 1, h: 1 },
      { i: "d", x: 1, y: 4, w: 1, h: 1 },
    ];

    return (
      <ResponsiveGridLayout
        {...this.props}
        isResizable
        rowHeight={200}
        measureBeforeMount
        layouts={{ lg: layout, md: layout, sm: layout }}
        breakpoints={{ lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 }}
        cols={{ lg: 4, md: 4, sm: 2, xs: 2, xxs: 1 }}>
        {
          layout.map((item) => <GridItem key={item.i}>{item.i}</GridItem>)
        }
      </ResponsiveGridLayout>
    );
  }
}

export default styled(GridLayout) `

  .react-grid-layout {
    position: relative;
    transition: height 200ms ease;
  }
  .react-grid-item {
    transition: all 200ms ease;
    transition-property: left, top;
  }
  .react-grid-item.cssTransforms {
    transition-property: transform;
  }
  .react-grid-item.resizing {
    z-index: ${index.grid};
    will-change: width, height;
  }

  .react-grid-item.react-draggable-dragging {
    transition: none;
    z-index: ${index.gridDragging};
    will-change: transform;
  }

  .react-grid-item.react-grid-placeholder {
    background: mediumpurple;
    opacity: 0.2;
    transition-duration: 100ms;
    z-index: ${index.gridPlaceholder};
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -o-user-select: none;
    user-select: none;
  }

  .react-grid-item > .react-resizable-handle {
    position: absolute;
    width: 20px;
    height: 20px;
    bottom: 0;
    right: 0;
    cursor: se-resize;
  }

  .react-grid-item > .react-resizable-handle::after {
    content: "";
    position: absolute;
    right: 3px;
    bottom: 3px;
    width: 5px;
    height: 5px;
    border-right: 2px solid rgba(0, 0, 0, 0.4);
    border-bottom: 2px solid rgba(0, 0, 0, 0.4);
  }
`;
