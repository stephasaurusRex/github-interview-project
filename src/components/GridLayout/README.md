GridLayout
===

A resizable grid layout, using [react-grid-layout](https://github.com/STRML/react-grid-layout).

Great for dashboards.

<!-- STORY -->
