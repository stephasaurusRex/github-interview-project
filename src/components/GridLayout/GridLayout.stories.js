import { storiesOf } from "@storybook/react";
import React from "react";
import { withDocs } from "storybook-readme";

import GridLayout from "./GridLayout";
import readme from "./README.md";

storiesOf("GridLayout", module)
  .addDecorator(withDocs(readme))
  .add("basic usage", () => <GridLayout />);
