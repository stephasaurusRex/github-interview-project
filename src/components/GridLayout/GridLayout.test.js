import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import GridLayout from "./GridLayout";

describe("GridLayout", () => {

  it("renders without props", () => {
    const tree = renderer.create(<GridLayout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
