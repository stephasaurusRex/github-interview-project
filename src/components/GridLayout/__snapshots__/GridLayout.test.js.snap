// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridLayout renders without props 1`] = `
.c0 .react-grid-layout {
  position: relative;
  -webkit-transition: height 200ms ease;
  transition: height 200ms ease;
}

.c0 .react-grid-item {
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  -webkit-transition-property: left,top;
  transition-property: left,top;
}

.c0 .react-grid-item.cssTransforms {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c0 .react-grid-item.resizing {
  z-index: 1;
  will-change: width,height;
}

.c0 .react-grid-item.react-draggable-dragging {
  -webkit-transition: none;
  transition: none;
  will-change: transform;
}

.c0 .react-grid-item.react-grid-placeholder {
  background: mediumpurple;
  opacity: 0.2;
  -webkit-transition-duration: 100ms;
  transition-duration: 100ms;
  z-index: 2;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -o-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .react-grid-item > .react-resizable-handle {
  position: absolute;
  width: 20px;
  height: 20px;
  bottom: 0;
  right: 0;
  cursor: se-resize;
}

.c0 .react-grid-item > .react-resizable-handle::after {
  content: "";
  position: absolute;
  right: 3px;
  bottom: 3px;
  width: 5px;
  height: 5px;
  border-right: 2px solid rgba(0,0,0,0.4);
  border-bottom: 2px solid rgba(0,0,0,0.4);
}

<div
  className="c0"
  style={undefined}
/>
`;
